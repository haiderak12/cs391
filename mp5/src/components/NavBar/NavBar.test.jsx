import renderer from 'react-test-renderer';
import { test, expect, vi } from 'vitest';
import NavBar from './NavBar.jsx';

test('The NavBar component should render', () => {
  const view = renderer
      .create(<NavBar />)
      .toJSON();
  expect(view).toMatchSnapshot();
});

test('The NavBar component should render with a title', () => {
  const title = 'Test application';
  const component = renderer
      .create(<NavBar title={title} />)
      .toJSON();
  expect(component).toMatchSnapshot();
});

test('The NavBar component should respond to button clicks', () => {
  const mockFunction = vi.fn();
  const component = renderer
      .create(<NavBar goBack={mockFunction} openForm={mockFunction} />);

  component.root.findByProps({ children: '< Go Back' }).props.onClick();
  expect(mockFunction).toHaveBeenCalled();

  component.root.findByProps({ children: '+ Add Review' }).props.onClick();
  expect(mockFunction).toHaveBeenCalledTimes(2);
});
