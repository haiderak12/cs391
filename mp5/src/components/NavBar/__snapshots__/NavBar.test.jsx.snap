// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The NavBar component should render 1`] = `
<div
  className="sc-guDLey gtpvTe"
>
  <h2
    className="sc-dmyCSP jbzSYl"
  />
</div>
`;

exports[`The NavBar component should render with a title 1`] = `
<div
  className="sc-guDLey gtpvTe"
>
  <h2
    className="sc-dmyCSP jbzSYl"
  >
    Test application
  </h2>
</div>
`;
